pkg/agent/openflow/feature_templates.go

```diff
@@ -0,0 +1,369 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package openflow
+
+import (
+	"antrea.io/antrea/pkg/agent/openflow/cookie"
+	binding "antrea.io/antrea/pkg/ovs/openflow"
+	"antrea.io/antrea/pkg/util/runtime"
+)
+
+// OVS pipelines in AntreaProxy are generated by a framework called Flexible Pipeline. There are some concepts in this
+// framework,
+//
+// +--------------+ +--------------+ +--------------+ +--------------+ +--------------+ +--------------+
+// | FeatureTable | | FeatureTable | | FeatureTable | | FeatureTable | | FeatureTable | | FeatureTable |
+// +--------------+ +--------------+ +--------------+ +--------------+ +--------------+ +--------------+
+//         \               |                /                |                 \               /
+//          \              |               /                 |                  \             /
+//           \             |              /                  |                   \           /
+//      +------------------------------------+        +--------------+       +--------------------+
+//      |               Stage                |        |     Stage    |       |        Stage       |
+//      +------------------------------------+        +--------------+       +--------------------+
+//              \                                          /    \                        /
+//               \                                        /      \                      /
+//                \                                      /        \                    /
+//              +------------------------------------------+    +--------------------------+
+//              |                 Feature                  |    |          Feature         |
+//              +------------------------------------------+    +--------------------------+
+//                                  |                    \                         /
+//                                  |                     \                       /
+//                                  |                      \                     /
+//                     +------------------------+        +-------------------------+
+//                     |        Pipeline        |        |         Pipeline        |
+//                     +------------------------+        +-------------------------+
+//
+// FeatureTable:
+// FeatureTable in Flexible Pipeline is like "iptables entry" in iptables. It is the basic unit to build OVS pipelines.
+//
+// Stage:
+// Stage in Flexible Pipeline is like "chain" in iptables. The order of Stages is fixed. A Stage can be empty or have
```

Comment:
I would suggest using more formal wording(counterpart, resemble...) to replace "is like".

Reply:
A Stage can be empty or have one or more FeatureTables. A Stage has zero or more FeatureTables.

Reply:
Thanks.